core/src/main/scala/chisel3/package.scala

```diff
@@ -412,4 +412,19 @@ package object chisel3 {
     "this feature will not be supported as part of the migration to the MLIR-based FIRRTL Compiler (MFC). For more information about this migration, please see the Chisel ROADMAP.md."
 
   final val deprecatedPublicAPIMsg = "APIs in chisel3.internal are not intended to be public"
+  
+  object BundleInit {
+    implicit class AddInitConstruct[T <: Record](bun: T) {
+      def Init(fs: (T => (Data, Data))*): T = {
```

Comment:
```suggestion
      def Init(fs: (T => (Data, Data))*)(implicit sourceInfo: SourceInfo): T = {
```
As I alluded to in my comment, https://github.com/chipsalliance/chisel/pull/3060#issuecomment-1452793826, you need to thread a source locator through so that it can be materialized in the caller's code.